scripts/reportInstalls.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { promisify } from 'util';
+import { exec } from 'child_process';
+import * as appInsights from 'applicationinsights';
+import { DEFAULT_AIKEY } from '../packages/salesforcedx-utils-vscode/src/constants';
+
+const promisifiedExec = promisify(exec);
+
+interface InstallTelemetryData {
+  // To align with the properties property for appInsightsClient.trackEvent
+  [key: string]: string;
+  install: string;
+  ratingcount: string;
+  marketplace: string;
+  'common.extname': string;
+}
+
+// These properties match the values provided in the statistics array from the
+// vsce cli output
+interface InstallTelemetryDataMS extends InstallTelemetryData {
+  averagerating: string;
+  trendingdaily: string;
+  trendingmonthly: string;
+  trendingweekly: string;
+  updateCount: string;
+  weightedRating: string;
+  downloadCount: string;
+}
+
+type TelemetryData = InstallTelemetryData | InstallTelemetryDataMS;
+
+// Set to true for local development
+const DRY_RUN = false;
+const extensionsToTrack = [
+  'salesforcedx-vscode-core',
```

Comment:
Probably a selfish ask, but can you put gpt first? I think that's the one that has the highest visibility, so if we're reporting them at all, we should make sure that that one goes in.